src/app/shared/models/action-trigger.ts

```diff
@@ -1,5 +1,43 @@
+import { KeyAlias } from 'src/app/keyboard/value-objects/key-alias'
+
 export interface ActionTrigger<Data = unknown> {
   id: string
   actionId: string
   data: Data
 }
+
+export interface KeyboardTriggerData {
+  keys: string[]
+  actionArguments?: unknown
+}
+
+export interface KeyboardAndSelectionTriggerData extends KeyboardTriggerData {
+  selected?: boolean
+}
+
+export interface CreateActionTrigger<Data = unknown> {
+  actionId: string
+  data: Data
+}
+
+export enum ActionTriggerSortKeys {
+  ACTION_ID_A_Z = 'ACTION_ID_A_Z',
+  ACTION_ID_Z_A = 'ACTION_ID_Z_A',
+  SHORTCUT_A_Z = 'SHORTCUT_A_Z',
+  SHORTCUT_Z_A = 'SHORTCUT_Z_A',
+}
+
+export enum UserActionsWithSelectedTriggers {
+  TOGGLE_SELECT = 'TOGGLE_SELECT',
+  EXPORT = 'EXPORT',
+  DELETE = 'DELETE',
+}
+
+export const SHORTCUT_KEYS_MAPPINGS: Record<string, string> = {
+  Enter: KeyAlias.ANY_ENTER,
+  NumpadEnter: KeyAlias.ANY_ENTER,
+  ShiftLeft: KeyAlias.SHIFT,
+  ShiftRight: KeyAlias.SHIFT,
+  ControlLeft: KeyAlias.CONTROL,
+  ControlRight: KeyAlias.CONTROL,
+}
```

Comment:
Your are in a file called `action-trigger.ts`.
Most of what you added here is keyboard specific and should have a its own file. ActionTrigger is not keyboard specific (that's just what we have paired it with so far).

Reply:
Done